apps/sms/test/marionette/lib/views/conversation/view.js

```diff
@@ -60,6 +89,114 @@ ConversationView.prototype = {
     return this._switchToContactsPageObject();
   },
 
+  openParticipants: function() {
+    this.accessors.headerTitle.tap();
+
+    var ParticipantsView = require('../participants/view');
+    var participantsView = new ParticipantsView(this.client);
+    participantsView.accessors.waitToAppear();
+    return participantsView;
+  },
+
+  openReport: function(messageId) {
+    var message = this.accessors.findMessage(messageId);
+
+    // Show context menu on message bubble.
+    this.client.loader.getActions().longPress(message, 1 /* sec */).perform();
+    this.menuAccessors.selectAppMenuOption('View message report');
+
+    var ReportView = require('../report/view');
+    var reportView = new ReportView(this.client);
+    reportView.accessors.waitToAppear();
+    return reportView;
+  },
+
+  deleteMessage: function(messageId) {
+    var lastMessageWillBeDeleted = this.messages.length === 1;
+
+    var message = this.accessors.findMessage(messageId);
+
+    // Show context menu on message bubble.
+    this.client.loader.getActions().longPress(message, 1 /* sec */).perform();
+    this.menuAccessors.selectAppMenuOption('Delete');
+
+    var dialogView = new (require('../dialog/view'))(this.client);
+    dialogView.chooseAction('Delete');
+
+    // If deleted message was the last one, let's return to Inbox.
+    if (lastMessageWillBeDeleted) {
+      var inboxView = new (require('../inbox/view'))(this.client);
+      inboxView.accessors.waitToAppear();
+
+      return inboxView;
+    }
+
+    return this;
```

Comment:
Cause this method can forward user to a different view (Inbox), similar to ```goToInbox```,  ```return this;``` (returns current view) is for consistency, so that consumer can always expect view object as the result of this method.

What do you think?